custom_metrics/privacy.js

```diff
@@ -15,10 +46,264 @@ return JSON.stringify({
       a.innerText.match(pattern)
     ).length;
   })(),
-  iab_tcf:
-    // IAB Transparency and Consent Framework is integrated on the page.
-    // docs v1: https://github.com/InteractiveAdvertisingBureau/GDPR-Transparency-and-Consent-Framework
-    // docs v2: https://github.com/InteractiveAdvertisingBureau/GDPR-Transparency-and-Consent-Framework/blob/master/TCFv2
-    (typeof window.__cmp == 'function') |
-    (typeof window.__tcfapi == 'function'),
+
+  // Consent Management Platforms
+
+  /**
+   * IAB Transparency and Consent Framework v1
+   * https://github.com/InteractiveAdvertisingBureau/GDPR-Transparency-and-Consent-Framework/blob/master/CMP%20JS%20API%20v1.1%20Final.md
+   */
+  iab_tcf_v1: (() => {
+    let consentData = {
+      present: typeof window.__cmp == 'function',
+      data: null,
+      compliant_setup: null,
+    };
+    // description of `__cmp`: https://github.com/InteractiveAdvertisingBureau/GDPR-Transparency-and-Consent-Framework/blob/master/CMP%20JS%20API%20v1.1%20Final.md#what-api-will-need-to-be-provided-by-the-cmp-
+    if (consentData.present) {
+      // Standard command: 'getVendorConsents'
+      // cf. https://github.com/InteractiveAdvertisingBureau/GDPR-Transparency-and-Consent-Framework/blob/master/CMP%20JS%20API%20v1.1%20Final.md#what-api-will-need-to-be-provided-by-the-cmp-
+      // Test site: ?
+      window.__cmp('getVendorConsents', null, (result, success) => {
```

Comment:
Executing application code from the custom metric can get weird.

Does this function have any side-effects on the page? We might want to avoid changing the page, to avoid tainting other metrics.
Does it necessarily support this function signature? We should wrap it in a try/catch in case there are any incompatibilities.

Reply:
It's the only way to obtain the necessary data. Since they're (marked as) pure getters (in the documentation), I wouldn't expect any side effects.

I'll get to the try/catch, even though any library implementing this API is supposedly [required to support these methods](https://github.com/InteractiveAdvertisingBureau/GDPR-Transparency-and-Consent-Framework/blob/master/TCFv2/IAB%20Tech%20Lab%20-%20CMP%20API%20v2.md#how-does-the-cmp-provide-the-api).

Reply:
done in 74b46c6